src/App.tsx

```diff
@@ -17,18 +34,39 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                currentQuery={currentQuery}
+                onFilterChange={setFilteringOptions}
+                onInputChange={setCurrentQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0
+                ? (
+                  <TodoList
+                    currentQuery={currentQuery}
+                    filteringOptions={filteringOptions}
+                    todoItems={todos}
+                    onSelectTodo={setSelectedTodoId}
+                    onUserSelect={setSelectedUserId}
+                    selectedTodoId={selectedTodoId}
+                  />
+                )
+                : <Loader />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedUserId && (
+        <TodoModal
+          todos={todos}
+          selectedTodoId={selectedTodoId}
+          selectedUserId={selectedUserId}
```

Comment:
Let's store in the state a selectedTodo, not `selectedTodoId`, `selectedUserId` separatly

Reply:
I have created new state for selectedTodo, but I`m still using IDs for recognizing which one is selected to apply styles